public/components/common/seach/queryBar.tsx

```diff
@@ -9,42 +9,166 @@
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'eval',
+    description: "The eval command evaluate the expression and append the result to the search result."},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'fields',
+    description: "Using field command to keep or remove fields from the search result."},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'head',
+    description: "The head command returns the first N number of specified results in search order."},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'rare',
+      description: "Using rare command to find the least common tuple of values of all fields in the field list."},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'rename',
+    description: "Using rename command to rename one or more fields in the search result."},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'sort',
+    description: "Using sort command to sorts all the search result by the specified fields."},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'stats',
+    description: "Using stats command to calculate the aggregation from search result."},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'top',
+    description: "Using top command to find the most common tuple of values of all fields in the field list."},
+  {type: {iconType: 'search', color: 'tint1'} ,label: 'where',
+    description: "The where command bool-expression to filter the search result."},
+]
+
+const indices: EuiSuggestItemProps[] = [
+  {type: {iconType: 'kqlField', color: 'tint4'} ,label: 'source=opensearch_dashboards_sample_data_ecommerce'},
+  {type: {iconType: 'kqlField', color: 'tint4'} ,label: 'source=opensearch_dashboards_sample_data_flights'},
+  {type: {iconType: 'kqlField', color: 'tint4'} ,label: 'source=opensearch_dashboards_sample_data_logs'}
+]
 
+export function QueryBar(props: IQueryBarProps) {
+  
   const {
     query,
     handleQueryChange,
     handleQuerySearch
   } = props;
 
+  // State to keep track of current input
+  // Query contains the queries to send to backend, inputValue just keeps track of what's
+  // in the input bar at the time. This ensures that the query doesn't disappear when we
+  // switch tabs or hit enter.
+  const [inputValue, setInputValue] = useState(query[ RAW_QUERY ]);
+
+  const splittedModel = inputValue.split(' ');
+  const prefix = splittedModel[splittedModel.length - 1];
+
+  // Function to grab suggestions
+  const getSuggestions = () => {
+    // First commands should either be search or source
+    // source should be followed by an available index
+    if (splittedModel.length === 1) {
+      if (prefix.startsWith("source=")) {
+        let str = prefix.replace("source=", "");
+        return getIndices(str);
+      }
+      return firstCommand;
+    }
+    // TODO: (Grammar implementation) Get commands based on grammar from backend
+    // Contextual suggestions are hardcoded for now
+    if (splittedModel.length > 1) {
```

Comment:
Can we make this second length check an `else if` block?

Reply:
Same with the `if` blocks within this conditional

Reply:
will do